src/Test/Perf/runner.csx

```diff
@@ -9,40 +10,64 @@ using System;
 
 InitUtilities();
 
-var myDir = MyWorkingDirectory();
-var skip = new HashSet<string> {
-    Path.Combine(myDir, "runner.csx"),
-    Path.Combine(myDir, "bootstrap.csx"),
-    Path.Combine(myDir, "util"),
-    Path.Combine(myDir, "infra"),
-};
+var testDirectory = PerfTestDirectory();
 
 var allResults = new List<Tuple<string, List<Tuple<int, string, object>>>>();
 var failed = false;
 
+// Run DownloadCPC before using the TraceManager because TraceManager uses the CPC downloaded binaries 
+DownloadCPC();
+var traceManager = new TraceManager();
+
 // Print message at startup
 Log("Starting Performance Test Run");
 Log("hash: " + StdoutFrom("git", "show --format=\"%h\" HEAD --").Split(new[] {"\r\n", "\r", "\n"}, StringSplitOptions.None)[0]);
 Log("time: " + DateTime.Now.ToString());
 
-// Run all the scripts that we've found and populate allResults.
-foreach (var script in AllCsiRecursive(myDir, skip)) {
-    var scriptName = Path.GetFileNameWithoutExtension(script);
-    Log("\nRunning " + scriptName);
-    try
-    {
-        var state = await RunFile(script);
-        var metrics = (List<Tuple<int, string, object>>) state.GetVariable("Metrics").Value;
-        allResults.Add(Tuple.Create(scriptName, metrics));
-    }
-    catch (Exception e)
+// This is an assumption. We can modify it later
```

Comment:
I don't think that we want the outer loop to be in control of iterations at all.